### PR TITLE
Warn of possible endstop/DIAG homing conflicts

### PR DESCRIPTION
**Pull request by thisiskeithb**
### Description

It is *extremely common* to troubleshoot homing issues where users do not remove diag jumpers from the motherboard or do not remove diag pins from drivers when using real endstops, so this PR adds a warning to remind users to do so.

I split the warning into `USES_DIAG_PINS` and `USES_DIAG_JUMPERS` flags to provide a clearer warning, but they could just as easily be combined and more generalized.

### Benefits

Users will be alerted to potential homing issues when their motherboard supports sensorless homing via diag signal and they are using real endstops.

### Related Issues
- Many in this issue queue, on Facebook, and on Discord. Some direct links:
- https://github.com/MarlinFirmware/Marlin/issues/21757
- https://github.com/MarlinFirmware/Marlin/issues/20295
- https://github.com/MarlinFirmware/Marlin/issues/16696

### CODE REVIEW

File: Marlin/src/inc/Warnings.cpp [hunk 1]
```diff
@@ -488,3 +488,12 @@
...
   #error "Z_SAFE_HOMING is recommended when homing with a probe. Enable Z_SAFE_HOMING or comment out this line to continue."
 #endif
+
+//
+// Warn users of potential endstop/DIAG pin conflicts to prevent homing issues when not using sensorless homing
+//
+#if !USE_SENSORLESS && USES_DIAG_JUMPERS
+  #warning "Motherboard DIAG jumpers must be removed when SENSORLESS_HOMING is disabled."
+#elif !USE_SENSORLESS && USES_DIAG_PINS
+  #warning "Driver DIAG pins must be removed when SENSORLESS_HOMING is disabled."
```
Comment (thinkyhead) [inline Marlin/src/inc/Warnings.cpp]: Can pins be removed with needle-nose pliers or are flush-cutters better?

Comment (thisiskeithb) [reply Marlin/src/inc/Warnings.cpp]: > Can pins be removed with needle-nose pliers or are flush-cutters better?

A soldering iron would be best, but cutters included with your printer work fine.

Comment (thinkyhead) [reply Marlin/src/inc/Warnings.cpp]: Hmm. Users might wonder why they need to physically remove pins entirely, especially if there is nothing connected to them.

Comment (thisiskeithb) [reply Marlin/src/inc/Warnings.cpp]: I think an explainer on TMC diag pin->endstop wiring might be a bit long for a warning, but pointing to various user manuals might work?

[Here's one from BTT](https://github.com/bigtreetech/BIGTREETECH-TMC2209-V1.2/blob/master/TMC2209%20problems%20solving.pdf).

**Pull request merged**